packages/vscode-messenger-common/src/messages.ts

```diff
@@ -176,8 +176,113 @@ export type NotificationHandler<P> = (params: P, sender: MessageParticipant) =>
  * Base API for Messenger implementations.
  */
 export interface MessengerAPI {
-    sendRequest<P, R>(type: RequestType<P, R>, receiver: MessageParticipant, params?: P): Promise<R>
+    sendRequest<P, R>(type: RequestType<P, R>, receiver: MessageParticipant, params?: P, cancelable?: CancellationToken): Promise<R>
```

Comment:
Let's use consistent parameter names for the CancellationTokens (I don't care much which one).

Reply:
Done